keras/layers/core.py

```diff
@@ -119,7 +119,7 @@ def __call__(self, X, mask=None, train=False):
         if tmp_previous is not None:
             self.set_previous(tmp_previous, False)
         else:
-            self.clear_previous(False)
+            del self.previous
```

Comment:
This will break for container layers. You need to use the `clear_previous` method so that containers can override the inherited implementation.